### PR TITLE
[FIX] l10n_fr_pos_cert: 'account.bank.statement' object has no attribute state

### PR DESCRIPTION
**Pull request by niyasraphy**
before this commit, in the code it is referenced to non existing field state in account.bank.statement model and thus it raise exception.

after this commit, the exception wont be raised and validation error will be done based on the state field state in account.bank.statement.line

issue: https://github.com/odoo/odoo/issues/114514

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_fr_pos_cert/models/account_bank_statement.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def _unlink_except_created_by_pos(self):
         for statement in self:
             if not statement.company_id._is_accounting_unalterable() or not statement.journal_id.pos_payment_method_ids:
                 continue
-            if statement.state != 'open':
+            if any(line.state != 'draft' for line in statement.line_ids):
```
Comment (smetl) [inline addons/l10n_fr_pos_cert/models/account_bank_statement.py:16]: Remove the condition. It's not relevant anymore

File: addons/l10n_fr_pos_cert/models/account_bank_statement.py [hunk 2]
```diff
@@ -27,5 +27,5 @@ def _unlink_except_created_by_pos(self):
             statement = st_line.statement_id
             if not statement.company_id._is_accounting_unalterable() or not statement.journal_id.pos_payment_method_ids:
                 continue
-            if statement.state != 'open':
+            if st_line.state != 'draft':
```
Comment (smetl) [inline addons/l10n_fr_pos_cert/models/account_bank_statement.py:30]: this is redundant with the existing check in point_of_sale I think
```py
    pos_session_id = fields.Many2one('pos.session', string="Session", copy=False)

    @api.ondelete(at_uninstall=False)
    def _unlink_except_linked_to_pos_session(self):
        for bsl in self:
            if bsl.pos_session_id:
                raise UserError(_("You cannot delete a bank statement line linked to Point of Sale session."))

```
This check can be removed imo

File: addons/l10n_fr_pos_cert/models/account_bank_statement.py (hunk 1 above)
Comment (niyasraphy) [reply addons/l10n_fr_pos_cert/models/account_bank_statement.py]: hi @smetl ,
thanks for the review and feedback, if this is the case, the entire file can be dropped as there is no other codes other than this checking.

let me know your suggestion

**Comment by smetl** [comment]
@niyasraphy Drop the whole file :) Thanks!

**Comment by niyasraphy** [comment]
@smetl updated the pr

**Comment by rhe-odoo** [comment]
@robodoo r+

**Pull request closed without merging**